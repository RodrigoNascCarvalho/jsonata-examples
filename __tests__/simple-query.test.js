const { testJsonataExpr } = require('./jsonata-helper')
const inputJson = require('example-json')

describe('Simple Queries Examples', () => {
    test('SQ01: Get descriptions for each product', () => {
        testJsonataExpr(inputJson, 'Account.Order.Product.Description')
    })
    
    test('SQ02: Flattening of data, select colors for description', () => {
        testJsonataExpr(inputJson, 'Account.Order.Product.Description.Colour')
    })
    
    test('SQ03: Flattening of data, select only first product of each order', () => {
        testJsonataExpr(inputJson, 'Account.Order.Product[0]')
    })
    
    test('SQ04: Flattening of data, select only first color', () => {
        testJsonataExpr(inputJson, '(Account.Order.Product.Description.Colour)[0]')
    })
    
    test('SQ05: Flattening of data, select only second color', () => {
        testJsonataExpr(inputJson, '(Account.Order.Product.Description.Colour)[1]')
    })
    
    test('SQ06: Flattening of data, select only last color', () => {
        testJsonataExpr(inputJson, '(Account.Order.Product.Description.Colour)[-1]')
    })
    
    test('SQ07: Flattening of data, select only before last color', () => {
        testJsonataExpr(inputJson, '(Account.Order.Product.Description.Colour)[-2]')
    })
    
    test('SQ08: Flattening of data, select only first three colors', () => {
        testJsonataExpr(inputJson, '(Account.Order.Product.Description.Colour)[[0..2]]')
    })
})