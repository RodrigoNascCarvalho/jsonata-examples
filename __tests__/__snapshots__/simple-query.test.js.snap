// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simple Queries Examples SQ01: Get descriptions for each product 1`] = `
Array [
  Object {
    "Colour": "Purple",
    "Depth": 210,
    "Height": 200,
    "Weight": 0.75,
    "Width": 300,
  },
  Object {
    "Colour": "Orange",
    "Depth": 210,
    "Height": 200,
    "Weight": 0.6,
    "Width": 300,
  },
  Object {
    "Colour": "Blue",
    "Depth": 210,
    "Height": 200,
    "Weight": 0.75,
    "Width": 300,
  },
  Object {
    "Colour": "Black",
    "Depth": 210,
    "Height": 20,
    "Weight": 2,
    "Width": 30,
  },
  Object {
    "Colour": "Yellow",
    "Depth": 210,
    "Height": 200,
    "Weight": 0.75,
    "Width": 300,
  },
  Object {
    "Colour": "Green",
    "Depth": 210,
    "Height": 100,
    "Weight": 2,
    "Width": 30,
  },
]
`;

exports[`Simple Queries Examples SQ02: Flattening of data, select colors for description 1`] = `
Array [
  "Purple",
  "Orange",
  "Blue",
  "Black",
  "Yellow",
  "Green",
]
`;

exports[`Simple Queries Examples SQ03: Flattening of data, select only first product of each order 1`] = `
Array [
  Object {
    "Description": Object {
      "Colour": "Purple",
      "Depth": 210,
      "Height": 200,
      "Weight": 0.75,
      "Width": 300,
    },
    "Price": 34.45,
    "Product Name": "Bowler Hat",
    "ProductID": 858383,
    "Quantity": 2,
    "SKU": "0406654608",
  },
  Object {
    "Description": Object {
      "Colour": "Blue",
      "Depth": 210,
      "Height": 200,
      "Weight": 0.75,
      "Width": 300,
    },
    "Price": 34.45,
    "Product Name": "Bowler Hat",
    "ProductID": 858383,
    "Quantity": 4,
    "SKU": "040657863",
  },
  Object {
    "Description": Object {
      "Colour": "Yellow",
      "Depth": 210,
      "Height": 200,
      "Weight": 0.75,
      "Width": 300,
    },
    "Price": 34.45,
    "Product Name": "Bowler Hat",
    "ProductID": 858383,
    "Quantity": 2,
    "SKU": "0406654608",
  },
]
`;

exports[`Simple Queries Examples SQ04: Flattening of data, select only first color 1`] = `"Purple"`;

exports[`Simple Queries Examples SQ05: Flattening of data, select only second color 1`] = `"Orange"`;

exports[`Simple Queries Examples SQ06: Flattening of data, select only last color 1`] = `"Green"`;

exports[`Simple Queries Examples SQ07: Flattening of data, select only before last color 1`] = `"Yellow"`;

exports[`Simple Queries Examples SQ08: Flattening of data, select only first three colors 1`] = `
Array [
  "Purple",
  "Orange",
  "Blue",
]
`;
